File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.java
class Solution {
    public int myAtoi(String s) {
        long ans = 0;
        boolean sign = true;
        int flag = 0;
        for(char ch : s.toCharArray()) {
            if(flag == 2) break;
            if(ch == ' ') {
                if(flag == 1) break;
                continue;
            }
            else if(ch == '-') {
                if(flag == 1) break;
                flag = 1;
                sign = false;
            }
            else if(ch == '+') {
                if(flag == 1) break;
                flag = 1;
                continue;
            }
            else if(ch >= '0' && ch <= '9') {
                flag = 1;
                if(sign) {
                    ans = (ans * 10) + (ch - '0');
                    if(ans > Integer.MAX_VALUE) {
                        return Integer.MAX_VALUE;
                    }
                }
                else {
                    ans = (ans * 10) - (ch - '0');
                    if(ans < Integer.MIN_VALUE) {
                        return Integer.MIN_VALUE;
                    }
                }
            }
            else {
                break;
            }
        }
        return (int) ans;
    }
}

File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.py
class Solution:
    def myAtoi(self, s: str) -> int:
        j = 0
        for i in range(len(s)):
            if s[i] == " ": continue
            else:
                j = i
                break
        sign = 1
        if j < len(s) and s[j] == "-":
            sign = -1
            j += 1
        elif j < len(s) and s[j] == "+":
            j += 1
        ans = 0
        for i in range(j, len(s)):
            if s[i].isdigit():
                ans = ans * 10 + ord(s[i]) - ord('0')
            else: break
        ans = ans * sign
        if ans > (2 ** 31) - 1: ans = 2 ** 31 - 1
        if ans < (-2 ** 31): ans = -2 ** 31
        return ans